src/components/CourseModal/SignupButton.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+
+import Button from '@mui/material/Button';
+import { ConfirmCard } from '@/components/ConfirmCard';
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+
+type Props = {
+  courseId: string;
+};
+
+export default function SignupButton({ courseId }: Props) {
+  const [backdropOpen, setBackdropOpen] = useState(false);
+  const router = useRouter();
+
+  const handleSignup = async () => {
+    try {
+      const response = await fetch('/api/course/signup', {
+        method: 'POST',
+        body: JSON.stringify(courseId),
+      });
+
+      if (!response.ok) {
+        throw response;
+      }
+      alert('Successfully signed up!');
+      router.refresh();
+    } catch (error: any) {
+      alert(error?.statusText ?? 'Internal server error');
+      console.error(error);
+    }
+  };
+
+  return (
+    <>
+      <Button
+        onClick={() => {
+          setBackdropOpen(true);
+        }}
+      >
+        Signup
```

Comment:
I would rather opt to use "Enroll" in terms of language choice here. It generally fits an educational setting better.

Reply:
I added `responsiveFontSizes` into the MUI theme that scales the header tags automatically. Breakpoint could be defined manually later if there is a need for more control.
https://mui.com/material-ui/customization/typography/#responsive-font-sizes

Reply:
Nice, excellent job. I tested and it's tons better now.

One thing I'm wondering though that as the responsive font sizes affect all typography elements, at least to my understanding, it might cause unexpected text formatting in UI elsewhere. Have to probably manually check that the other branches don't get unwanted behaviour with it. Other option would be perhaps to create a new heading type in theme and make that reponsive with few breakpoints. That way it would affect only this implementation. But basically, better probably to move on to merging instead. Can improve later.